lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -140,81 +183,66 @@ def test__has_access_descriptor(self):
-        mock_unit.start = datetime.datetime.now(pytz.utc) + datetime.timedelta(days=1)  # release date in the future
-        mock_unit.visible_to_staff_only = True
-        self.verify_access(mock_unit, False)
-
-        # Start date in the future, staff lock off.
-        mock_unit.visible_to_staff_only = False
-        self.verify_access(mock_unit, False)
+        self.assertTrue(bool(access._has_access_descriptor(
+            self.beta_user, 'load', mock_unit, course_key=self.course.course_key)))
 
+    @ddt.data(None, YESTERDAY, TOMORROW)
     @patch.dict('django.conf.settings.FEATURES', {'DISABLE_START_DATES': False})
     @patch('courseware.access.get_current_request_hostname', Mock(return_value='preview.localhost'))
-    def test__has_access_descriptor_in_preview_mode(self):
+    def test__has_access_descriptor_in_preview_mode(self, start):
         """
         Tests that descriptor has access in preview mode.
         """
         mock_unit = Mock(user_partitions=[])
         mock_unit._class_tags = {}  # Needed for detached check in _has_access_descriptor
-
-        # No start date.
         mock_unit.visible_to_staff_only = False
+        mock_unit.start = start
         self.verify_access(mock_unit, True)
 
-        # Start date in the past.
-        mock_unit.start = datetime.datetime.now(pytz.utc) - datetime.timedelta(days=1)
-        self.verify_access(mock_unit, True)
-
-        # Start date in the future.
-        mock_unit.start = datetime.datetime.now(pytz.utc) + datetime.timedelta(days=1)  # release date in the future
-        self.verify_access(mock_unit, True)
-
+    @ddt.data(
+        (None, True),
+        (YESTERDAY, True),
+        (TOMORROW, False, access_response.StartDateError)
+    )
+    @ddt.unpack
     @patch.dict('django.conf.settings.FEATURES', {'DISABLE_START_DATES': False})
     @patch('courseware.access.get_current_request_hostname', Mock(return_value='localhost'))
-    def test__has_access_descriptor_when_not_in_preview_mode(self):
+    def test__has_access_descriptor_when_not_in_preview_mode(self, start, expected_access, expected_error_type=None):
         """
         Tests that descriptor has no access when start date in future & without preview.
         """
         mock_unit = Mock(user_partitions=[])
         mock_unit._class_tags = {}  # Needed for detached check in _has_access_descriptor
-
-        # No start date.
         mock_unit.visible_to_staff_only = False
-        self.verify_access(mock_unit, True)
-
-        # Start date in the past.
-        mock_unit.start = datetime.datetime.now(pytz.utc) - datetime.timedelta(days=1)
-        self.verify_access(mock_unit, True)
-
-        # Start date in the future.
-        mock_unit.start = datetime.datetime.now(pytz.utc) + datetime.timedelta(days=1)  # release date in the future
-        self.verify_access(mock_unit, False)
+        mock_unit.start = start
+        self.verify_access(mock_unit, expected_access, expected_error_type)
```

Comment:
yeah, good call - my only worry was that its possible to get false without a specific type of error (aka ACCESS_DENIED)